include/extractor/intersection_bearings_container.hpp

```diff
@@ -46,9 +46,9 @@ template <storage::Ownership Ownership> class IntersectionBearingsContainer
     IntersectionBearingsContainer &operator=(IntersectionBearingsContainer &&) = default;
     IntersectionBearingsContainer &operator=(const IntersectionBearingsContainer &) = default;
 
-    IntersectionBearingsContainer(std::vector<BearingClassID> node_to_class_id_,
+    IntersectionBearingsContainer(const std::vector<BearingClassID> &node_to_class_id,
```

Comment:
What was the error message here? Here we guarantee a copy.
By having this as by-value, we can avoid a copy by doing two `move` operations. 

It looks like the code isn't currently taking advantage of this, but it could.
https://github.com/Project-OSRM/osrm-backend/blob/3d2db20777a06ec064876f6c7526f485e36209ea/src/extractor/extractor.cpp#L863

Reply:
It was `performance-unnecessary-value-param` it seems it is because this `template <typename T> using Vector = util::ViewOrVector<T, Ownership>;` does not have constructor which accepts `std::vector<...>&&`, but tbh I am not sure and decided to add `NOLINT...` for the time being to preserve a status quo here.

Reply:
I see. The underlying container will depend on the ownership, so it can end up being `std::vector`. clang-tidy might not be able to figure it out.

Reply:
Also, I can omit the linting comment on both my dev environments (Big Sur-clang14, Ubuntu 21.04-clang12) without error, so it does seem a bit unsure its relevance.

Reply:
> Also, I can omit the linting comment on both my dev environments (Big Sur-clang14, Ubuntu 21.04-clang12) without error, so it does seem a bit unsure of its relevance.

It doesn't fire for me locally too, but does fire on CI.

Reply:
Hmm, it must be a bug in that version. Even when I change the only invocation to be moving an r-value (see #6275), we get the same error. In any case, we'll need the nolint comment.